packages/react/src/editor/BlockNoteDefaultUI.tsx

```diff
@@ -15,8 +17,47 @@ export type BlockNoteDefaultUIProps = {
   tableHandles?: boolean;
 };
 
+
+
+
 export function BlockNoteDefaultUI(props: BlockNoteDefaultUIProps) {
+
+
+
+//this makes sure emjois aren't fetched at every render
+init({ Data })
+
+async function search(value: string) {
+  if(value == ''){
+    //Don't do unnecessary linear search if no search query
+    return Object.values(Data.emojis).map((emoji : any)=>(
+      emoji.skins[0].native
+    ))
+  }
+  const emojisToShow : any[] = []
+  //begin the linear search
+  Object.values(Data.emojis).forEach((emoji : any)=>{
```

Comment:
Maybe use this? https://github.com/missive/emoji-mart?tab=readme-ov-file#%EF%B8%8F%EF%B8%8F-headless-search

Reply:
thanks for reminding, I intended to use it but just forgot.